activerecord/test/cases/associations/required_test.rb

```diff
@@ -22,30 +22,6 @@ class Child < ActiveRecord::Base
     @connection.drop_table "children", if_exists: true
   end
 
-  test "belongs_to associations are not required by default" do
```

Comment:
Right. Maybe it is good to debug which value it is getting on this specific test and why not it was not failing.